2-api/server.js

```diff
@@ -0,0 +1,275 @@
+      [customerId]
+    )
+    .then((result) => {
+      res.status(200).json(result.rows);
+    })
+    .catch((err) => console.error(err));
+});
+
+app.put("/customers/:customerId", (req, res) => {
+  const customerId = req.params.customerId;
+  const updateName = req.body.name;
+  const updateAddress = req.body.address;
+  const updateCity = req.body.city;
+  const updateCountry = req.body.country;
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400).send("Customer not found");
+      } else {
+        pool.query(
+          `UPDATE customers SET name=$1, address=$2, city=$3, country=$4 WHERE id=${customerId}`,
+          [updateName, updateAddress, updateCity, updateCountry],
+          (err, result) => {
+            if (!err) {
+              res
+                .status(200)
+                .send("Customer successfully updated")
+                .json(result.rows);
+            } else {
+              console.error(err);
+            }
+          }
+        );
+      }
+    });
+});
+
+app.post("/customers/:customerId/orders", (req, res) => {
+  const customerId = req.params.customerId;
+  const newOrderDate = req.body.orderDate;
+  const newOrderRef = req.body.orderRef;
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400);
+      }
+    })
+    .query(
+      "INSERT INTO orders (customer_id, order_date, order_reference) VALUES ($1, $2, $3)",
+      [customerId, newOrderDate, newOrderRef],
+      (err, result) => {
+        if (!err) {
+          res.status(200).json(result.rows);
+        } else {
+          res.status(400);
+          console.error(err);
+        }
+      }
+    );
+});
+
+app.post("/customers", (req, res) => {
+  const newName = req.body.name;
+  const newAddress = req.body.address;
+  const newCity = req.body.city;
+  const newCountry = req.body.country;
+  if (!newName) {
+    return res.status(404).send("Please enter your name to proceed");
+  }
+  pool
+    .query("SELECT name FROM customers WHERE id=$1", [newName])
+    .then((result) => {
+      if (result.rows.length > 0) {
+        return res.status(404).send("Customer name already exists!");
+      } else {
+        pool.query(
+          "INSERT INTO customers (name, address, city, country) VALUES ($1, $2, $3, $4)",
+          [newName, newAddress, newCity, newCountry],
+          (err, result) => {
+            if (!err) {
+              res.status(200).send("Customer Added").json(result.rows);
+            } else {
+              res.status(400);
+              console.error(err);
+            }
+          }
+        );
+      }
+    });
+});
+
+app.delete("/customers/:customerId", (req, res) => {
+  const customerId = req.params.customerId;
+  pool
+    .query("DELETE FROM orders WHERE customer_id=$1", [customerId])
```

Comment:
You need to check if there are any orders - you can only delete a customer if they have no orders.